client/src/javascript/components/modals/settings-modal/DiskUsageTab.js

```diff
@@ -0,0 +1,134 @@
+import {Checkbox, Form, FormRow, Select, SelectItem, Radio} from 'flood-ui-kit';
+import {FormattedMessage, injectIntl} from 'react-intl';
+import React from 'react';
+
+import ModalFormSectionHeader from '../ModalFormSectionHeader';
+import SettingsStore from '../../../stores/SettingsStore';
+import SettingsTab from './SettingsTab';
+import SortableList from '../../general/SortableList';
+import DiskUsageStore from '../../../stores/DiskUsageStore';
+
+class DiskUsageTab extends SettingsTab {
+  tooltipRef = null;
+
+  state = {
+    diskItems: [],
+  };
+
+  componentWillMount() {
+    // assemble disk items from saved "mountPoints" and list of disks "disks"
+    const mountPoints = SettingsStore.getFloodSettings('mountPoints');
+    const disks = DiskUsageStore.getDiskUsage().reduce((a, c) => {
+      a[c.target] = c;
+      return a;
+    }, {});
+    const diskItems = [];
+    // first targets saved in mountPoints that exist in disks
+    diskItems.push(
+      ...mountPoints.filter(target => target in disks).map(target => ({
+        id: target,
+        visible: true,
+      })),
+    );
+    // then remaing targets from disks
+    diskItems.push(
+      ...Object.keys(disks)
+        .filter(target => !mountPoints.includes(target))
+        .map(target => ({
+          id: target,
+          visible: false,
+        })),
+    );
```

Comment:
I have a stylistic nit, and for posterity I'll document it. But don't feel obligated to implement it as it's trivial:

I would try to chain this, maybe something like:
```js
const diskItems = mountPoints
  .filter(target => target in disks)
  .map(target => ({
    id: target,
    visible: true,
  }))
  .concat(
    Object.keys(disks)
      .filter(target => !mountPoints.includes(target))
      .map(target => ({
        id: target,
        visible: false,
      }))
  );
```